data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java

```diff
@@ -49,26 +48,14 @@
   private final JsonProvider jsonProvider = new JsonProvider();
 
   private void initialise(DataProvider dataProvider, final int requestedPortNumber) {
-    initialise(
-        dataProvider.getChainStorageClient(),
-        dataProvider.getP2pNetwork(),
-        dataProvider.getCombinedChainDataClient(),
-        requestedPortNumber);
+    app.server().setServerPort(requestedPortNumber);
 
+    addBeaconHandlers(dataProvider);
+    addNetworkHandlers(new NetworkDataProvider(dataProvider.getP2pNetwork()));
```

Comment:
just pass dataProvider, then on the other side you'll be able to use dataProvider.getNetworkDataProvider()
we don't want to ever reference dataProvider.getP2pnetwork, that's why its deprecated, once we've done this change it's getting deleted...